rust/agama-dbus-server/src/network/model.rs

```diff
@@ -480,6 +583,76 @@ pub struct LoopbackConnection {
 }
 
 #[derive(Debug, Default, PartialEq, Clone)]
+pub struct BondConnection {
+    pub base: BaseConnection,
+    pub bond: BondConfig,
+}
+
+impl BondConnection {
+    pub fn find_port(&self, name: &str) -> Option<&Connection> {
+        self.bond.ports.iter().find(|c| c.interface() == name)
+    }
+
+    pub fn set_ports(&mut self, ports: Vec<Connection>) {
+        self.bond.ports = ports;
+    }
+
+    pub fn set_options<T>(&mut self, options: T) -> Result<(), NetworkStateError>
+    where
+        T: TryInto<BondOptions>,
+        <T as TryInto<BondOptions>>::Error: std::fmt::Debug,
+    {
+        self.bond.options = options
+            .try_into()
+            .map_err(|_e| NetworkStateError::InvalidBondOptions)?;
+        Ok(())
+    }
+}
+
+#[derive(Debug, Default, Clone, PartialEq)]
+pub struct BondOptions(pub HashMap<String, String>);
+
+impl TryFrom<String> for BondOptions {
+    type Error = NetworkStateError;
+
+    fn try_from(value: String) -> Result<Self, Self::Error> {
+        let mut options = HashMap::new();
+
+        for opt in value.split_whitespace() {
+            let (opt_key, opt_value) = opt.trim().split_once('=').unwrap();
+            options.insert(opt_key.to_string(), opt_value.to_string());
+        }
+
+        Ok(BondOptions(options))
+    }
+}
+
+impl fmt::Display for BondOptions {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let opts = &self
+            .0
+            .iter()
+            .map(|(key, value)| format!("{key}={value}"))
+            .collect::<Vec<_>>();
+
+        write!(f, "{}", opts.join(" "))
+    }
```

Comment:
Not enough rusty?

```suggestion
    let mut spacer = "";
    for (key, value) in &self.0 {
        write!(f, "{spacer}{key}={value}");
        spacer = " ";
    }
    Ok(())
```

Reply:
Well, I would say that it is more a Ruby-like approach :-) Both of them look expressive enough.

Reply:
I thought the later would have less implicit memory effort, as it writes directly to the stream. But there is not critical path and I don't have hard feelings here :-)